agent/src/softwarecontaineragent.cpp

```diff
@@ -20,225 +22,245 @@ SoftwareContainerAgent::~SoftwareContainerAgent()
-    if (valid) {
-        m_containers[containerID] = nullptr;
-    } else {
-        log_error() << "Invalid container ID " << containerID;
-    }
+	if (isIdValid(containerID)) {
+		m_containers.erase(containerID);
+		m_containerIdPool.push_back(containerID);
+	} else {
+		log_error() << "Invalid container ID " << containerID;
+	}
 }
 
 bool SoftwareContainerAgent::checkContainer(ContainerID containerID, SoftwareContainer *&container)
 {
-    bool valid = ((containerID < m_containers.size()) && (m_containers[containerID] != nullptr));
-    if (valid) {
-        container = m_containers[containerID].get();
-    } else {
-        log_error() << "Invalid container ID " << containerID;
-    }
-
-    return valid;
+	if (isIdValid(containerID)) {
+		container = m_containers[containerID].get();
+		return true;
+	} else {
+		log_error() << "Invalid container ID " << containerID;
+	}
+
+	return false;
 }
 
 ReturnCode SoftwareContainerAgent::readConfigElement(const json_t *element)
 {
-    bool wo;
-    if(!read(element, "enableWriteBuffer", wo)) {
-        log_debug() << "enableWriteBuffer not found";
-        m_softwarecontainerWorkspace->m_enableWriteBuffer = false;
-    } else {
-        if (wo == true) {
-            m_softwarecontainerWorkspace->m_enableWriteBuffer = true;
-        } else {
-            m_softwarecontainerWorkspace->m_enableWriteBuffer = false;
-        }
-    }
-    return ReturnCode::SUCCESS;
+	bool wo;
+	if(!read(element, "enableWriteBuffer", wo)) {
+		log_debug() << "enableWriteBuffer not found";
+		m_softwarecontainerWorkspace->m_enableWriteBuffer = false;
+	} else {
+		if (wo == true) {
+			m_softwarecontainerWorkspace->m_enableWriteBuffer = true;
+		} else {
+			m_softwarecontainerWorkspace->m_enableWriteBuffer = false;
+		}
+	}
+	return ReturnCode::SUCCESS;
 }
 
 bool SoftwareContainerAgent::parseConfig(const std::string &config)
 {
-    if (config.size() == 0) {
-        log_warning() << "No configuration interpreted";
-        return false;
-    }
-
-    json_error_t error;
-    json_t *root = json_loads(config.c_str(), 0, &error);
-
-    if (!root) {
-        log_error() << "Could not parse config: " << error.text;
-        log_error() << config;
-        return false;
-    }
-
-    if (json_is_array(root)) {
-        for(size_t i = 0; i < json_array_size(root); i++) {
-            json_t *element = json_array_get(root, i);
-            if (json_is_object(element)) {
-                if (isError(readConfigElement(element))) {
-                    log_error() << "Could not read config element";
-                    return false;
-                }
-            } else {
-                log_error() << "json configuration is not an object";
-                return false;
-            }
-        }
-    } else {
-        log_error() << "Root JSON element is not an array";
-        return false;
-    }
-
-    return true;
+	if (config.size() == 0) {
```

Comment:
This chunk seems to be indented wrong. Please review the whole commit according to the style guidelines.